File: app/src/main/java/com/decode/tumblr/model/PostObject.java
package com.decode.tumblr.model;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "post_table")
public class PostObject {

    @PrimaryKey(autoGenerate = true)
    private int id;

    @NonNull
    private String title;

    @NonNull
    private String imgUrl;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @NonNull
    public String getTitle() {
        return title;
    }

    public void setTitle(@NonNull String title) {
        this.title = title;
    }

    @NonNull
    public String getImgUrl() {
        return imgUrl;
    }

    public void setImgUrl(@NonNull String imgUrl) {
        this.imgUrl = imgUrl;
    }
}

File: app/src/main/java/com/decode/tumblr/data/db/model/PostObject.kt
package com.decode.tumblr.data.db.model

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize

@Entity(tableName = "post_table")
@Parcelize
data class PostObject(

        @PrimaryKey
        var id: Long = 0,

        var title: String = "",

        var photoId: Int = 0,

        @Ignore
        var photoObject: PhotoObject? = null

) : Parcelable
